eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientInstantiationCheck.java

```diff
@@ -54,40 +99,39 @@
     public void beginTree(DetailAST root) {
         hasServiceClientAnnotation = false;
         isAsync = false;
-        isImplPackage = false;
     }
 
     @Override
     public void visitToken(DetailAST token) {
-        if (isImplPackage) {
-            return;
-        }
-
         switch (token.getType()) {
-            case TokenTypes.PACKAGE_DEF:
-                String packageName = FullIdent.createFullIdent(token.findFirstToken(TokenTypes.DOT)).getText();
-                isImplPackage = packageName.contains(".implementation");
```

Comment:
By removing the package_def check, aren't you now running the rules on all implementation classes?

Reply:
No, I suppress the rule on implementation package instead of checks in rules.
By doing so will have a better performance. Because running the rule on all java files. and tree-traversal all abstract nodes on all files to check if the class is in the implementation package won't be better than not starting check at initially.